pandas/core/ops.py

```diff
@@ -657,17 +658,31 @@ def wrapper(self, other):
         if isinstance(other, pd.Series):
             name = _maybe_match_name(self, other)
 
-            other = other.reindex_like(self).fillna(False).astype(bool)
-            return self._constructor(na_op(self.values, other.values),
+            if other.dtype.kind not in ['u', 'i']:
+                other = other.reindex_like(self).fillna(False).astype(bool)
```

Comment:
the `other.reindex_like(self)` bit is repeated twice here -- can you consolidate it?